python/src/py/objint.c

```diff
@@ -462,3 +462,17 @@ const mp_obj_type_t mp_type_int = {
     .binary_op = mp_obj_int_binary_op,
     .locals_dict = (mp_obj_dict_t*)&int_locals_dict,
 };
+
+mp_obj_t mp_obj_new_int_via_str(const char* value, int base) {
```

Comment:
Same comment as mp_obj_new_float_via_str:
could it be replaced by `mp_parse_num_integer(value, strlen(value), 16, NULL)`?

Reply:
Like I said, I firstly need to get the length of the string until the first \x0